src/model/model.cpp

```diff
@@ -11,42 +14,90 @@
 
 #include "util/assert.hpp"
 
-using namespace deepworks::graph;
+#include <iostream>
+#include <fstream>
+
+namespace dw = deepworks;
+using namespace dw::graph;
+
+struct is_data {
+   bool operator()(ade::NodeHandle nh) {
+       return tgraph.metadata(nh).get<Type>().t == Type::DATA;
+   }
+
+   TypedGraph& tgraph;
+};
+
+struct is_op {
+   bool operator()(ade::NodeHandle nh) {
+       return tgraph.metadata(nh).get<Type>().t == Type::OP;
+   }
+
+   TypedGraph& tgraph;
```

Comment:
Can we move this reference to operator arguments?

Reply:
t's done in order to reduce number of codes